lib/Newsletter/Scheduler/Scheduler.php

```diff
@@ -24,6 +24,25 @@ class Scheduler {
   const INTERVAL_MONTHLY = 'monthly';
   const INTERVAL_NTHWEEKDAY = 'nthWeekDay';
 
+  static function transitionHook($new_status, $old_status = null, $post = null) {
+    if(($new_status !== 'publish') || !($post instanceof \WP_Post)) {
```

Comment:
With fixed parameter in add filter $post and $old_status should be always defined

Reply:
I think you should also check here if the $post has one of types from WPPosts::getTypes()